runner/kafka/runner.go

```diff
@@ -81,8 +81,8 @@ func NewRunner(c Config, since time.Time, collector instrument.Collector) *Runne
 	return &Runner{
 		config:    c,
 		since:     since,
-		stopChan:  make(chan bool),
-		exitChan:  make(chan bool),
+		stopChan:  make(chan bool, 1),
+		exitChan:  make(chan bool, 1),
```

Comment:
changing to buffered channel will make the Close() function non blocking.
Initially the Close() was designed to be blocking waiting for all components to be flushed.

Reply:
The Runner.Close() function still receives from exitChan: https://github.com/etf1/kafka-message-scheduler/blob/c22c1266ff4da8094994b65b6e58aa7230ff52e5/runner/kafka/runner.go#L92

So AFAIK the Close() function should still be blocking?  That line won't finish until the Start() function writes to exitChan.  Otherwise, there is nothing to receive and it will wait & shouldn't matter if the channel is buffered or not.